groups/bsl/bslstl/bslstl_optional.h

```diff
@@ -765,787 +1431,10708 @@ void Optional_DataImp<TYPE>::emplace(
+    : d_allocator(allocator)
+        // If specified 'rhs' contains a value, initialize the 'value_type'
+        // object by move construction from '*rhs'.  Otherwise, create a
+        // disengaged 'optional'.  Use the specified 'allocator' to supply
+        // memory.
+    {
+        // Must be in-place inline because the use of 'enable_if' will
+        // otherwise break the MSVC 2010 compiler.
+        optional<ANY_TYPE>& lvalue = rhs;
+
+        if (lvalue.has_value()) {
+            emplace(MoveUtil::move(lvalue.value()));
+        }
+    }
+
+#else
+    template <class ANY_TYPE>
+    explicit optional(
+        bsl::allocator_arg_t,
+        allocator_type                                      allocator,
+        BloombergLP::bslmf::MovableRef<optional<ANY_TYPE> > rhs
+        BSLSTL_OPTIONAL_ENABLE_IF_CONSTRUCT_FROM_OPTIONAL_RVAL)
+    : d_allocator(allocator)
+        // If specified 'rhs' contains a value, initialize the 'value_type'
+        // object by move construction from '*rhs'.  Otherwise, create a
+        // disengaged 'optional'.  Use the specified 'allocator' to supply
+        // memory.
+    {
+        // Must be in-place inline because the use of 'enable_if' will
+        // otherwise break the MSVC 2010 compiler.
+        optional<ANY_TYPE>& lvalue = rhs;
+
+        if (lvalue.has_value()) {
+            emplace(MoveUtil::move(lvalue.value()));
+        }
+    }
+
+#endif  //BSLS_COMPILERFEATURES_SUPPORT_RVALUE_REFERENCES
+
+#ifdef BSLS_LIBRARYFEATURES_HAS_CPP17_BASELINE_LIBRARY
+    template <class ANY_TYPE = TYPE>
+    explicit optional(bsl::allocator_arg_t,
+                      allocator_type                 allocator,
+                      const std::optional<ANY_TYPE>& rhs
+        BSLSTL_OPTIONAL_ENABLE_IF_CONSTRUCT_FROM_STD_OPTIONAL_LVAL)
+    : d_allocator(allocator)
+        // If specified 'rhs' contains a value, initialize the 'value_type'
+        // object with '*rhs'.  Otherwise, create a disengaged 'optional'.  Use
+        // the specified 'allocator' to supply memory.
+    {
+        // Must be in-place inline because the use of 'enable_if' will
+        // otherwise break the MSVC 2010 compiler.
+        if (rhs.has_value()) {
+            this->emplace(rhs.value());
+        }
+    }
+
+    template <class ANY_TYPE = TYPE>
+    explicit optional(bsl::allocator_arg_t,
+                      allocator_type            allocator,
+                      std::optional<ANY_TYPE>&& rhs
+        BSLSTL_OPTIONAL_ENABLE_IF_CONSTRUCT_FROM_STD_OPTIONAL_RVAL)
+    : d_allocator(allocator)
+        // If specified 'rhs' contains a value, initialize the 'value_type'
+        // object by move construction from '*rhs'.  Otherwise, create a
+        // disengaged 'optional'.  Use the specified 'allocator' to supply
+        // memory.
+    {
+        // Must be in-place inline because the use of 'enable_if' will
+        // otherwise break the MSVC 2010 compiler.
+        std::optional<ANY_TYPE>& lvalue = rhs;
```

Comment:
No need for lvalue here.

Reply:
removed and applied the 'use std instead of Moveutil in C++11 land' comment to all of C++17 land aswell